### PR TITLE
[Doc] add explanations for entry filter

### PR DESCRIPTION
**Pull request by Anonymitaet**
Add docs for https://github.com/apache/pulsar/pull/12869 and https://github.com/apache/pulsar/pull/12970

Thanks for your tech inputs and guidance @315157973 

image

Preview looks good:
image

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@315157973 @Huanli-Meng @codelipenghui can you help review? Thanks

File: site2/docs/develop-plugin.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+---
+id: develop-plugin
+title: Pulsar plugin development
+sidebar_label: Plugin
+---
+
+You can develop various plugins for Pulsar, such as entry filters, protocol handlers, interceptors, and so on.
+
+## Entry filter
+
```
Comment (Huanli-Meng) [inline site2/docs/develop-plugin.md:10]: There should be a leading sentence between level 2 and level 3 headings.

File: site2/docs/develop-plugin.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+### What is an entry filter?
+
+The entry filter is an extension point for implementing a custom message entry strategy. With this implementation, you can decide **whether to send messages to consumers** (brokers can use the return values of entry filters to determine whether the messages need to be sent or discarded) or **send messages to specific target consumers.**
```
Comment (Huanli-Meng) [inline site2/docs/develop-plugin.md:13]: I think we can use 
```suggestion
The entry filter is an extension point for implementing a custom message entry strategy. With an entry filter, you can decide **whether to send messages to consumers** (brokers can use the return values of entry filters to determine whether the messages need to be sent or discarded) or **send messages to specific consumers.** 
```

File: site2/docs/develop-plugin.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+To implement features such as tagged messages or custom delayed messages, use [`subscriptionProperties`](https://github.com/apache/pulsar/blob/ec0a44058d249a7510bb3d05685b2ee5e0874eb6/pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L174), [`​​properties`](https://github.com/apache/pulsar/blob/ec0a44058d249a7510bb3d05685b2ee5e0874eb6/pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L533), and entry filters.
+
+### How to use an entry filter?
+
+Follow the steps below:
+
+1. Create a maven project.
```
Comment (Huanli-Meng) [inline site2/docs/develop-plugin.md:21]:
```suggestion
1. Create a Maven project.
```

File: site2/docs/develop-plugin.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+   
+2. Implement the `EntryFilter` interface.
+   
+3. Package the implementation class into NAR.
+
+4. Configure broker.conf and restart broker.
```
Comment (Huanli-Meng) [inline site2/docs/develop-plugin.md:27]:
```suggestion
4. Configure the `broker.conf` file and restart the broker.
```

File: site2/docs/develop-plugin.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+### How to use an entry filter?
+
+Follow the steps below:
+
+1. Create a maven project.
+   
+2. Implement the `EntryFilter` interface.
+   
+3. Package the implementation class into NAR.
```
Comment (Huanli-Meng) [inline site2/docs/develop-plugin.md:25]:
```suggestion
3. Package the implementation class into a NAR package.
```

File: site2/docs/develop-plugin.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    ```xml
+    <dependency>
+    <groupId>org.apache.pulsar</groupId>
+    <artifactId>pulsar-broker</artifactId>
+    <version>${pulsar.version}</version>
+    <scope>provided</scope>
+    </dependency>
+    ```
+
+2. Implement the  [`FilterResult filterEntry(Entry entry, FilterContext context);` method](https://github.com/apache/pulsar/blob/2adb6661d5b82c5705ee00ce3ebc9941c99635d5/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/plugin/EntryFilter.java#L34).
```
Comment (Huanli-Meng) [inline site2/docs/develop-plugin.md:46]:
```suggestion
2. Implement the [`FilterResult filterEntry(Entry entry, FilterContext context);` method](https://github.com/apache/pulsar/blob/2adb6661d5b82c5705ee00ce3ebc9941c99635d5/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/plugin/EntryFilter.java#L34).
```

File: site2/docs/develop-plugin.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+   - If the method returns `ACCEPT` or NULL, this message is sent to consumers.
+
+   - If the method returns `REJECT`, this message is filtered out and it does not consume message permits.
```
Comment (Huanli-Meng) [inline site2/docs/develop-plugin.md:50]: what's meaning of "consume message permits"

File: site2/docs/develop-plugin.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+   - If the method returns `REJECT`, this message is filtered out and it does not consume message permits. 
+
+   - If there are multiple entry filters, this message passes through all filters in the pipeline in a round-robin manner. If any entry filter returns `REJECT`, this message is discarded.
+
+    You can get entry metadata, subscriptions, and other information through `FilterContext`.
+
+3. Describe a NAR file.
+
+    Create an `entry_filter.yaml` file in the `resources/META-INF/services` directory to describe NAR.
```
Comment (Huanli-Meng) [inline site2/docs/develop-plugin.md:58]:
```suggestion
    Create an `entry_filter.yaml` file in the `resources/META-INF/services` directory to describe a NAR file.
```

File: site2/docs/develop-plugin.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+
+2. Generate a NAR file in the `target` directory.
+
+    ```script
+    mvn clean install
+    ```
+
+#### Step 4: configure and restart broker
+
+1. Configure the following parameters in the `broker.conf` file.
```
Comment (Huanli-Meng) [inline site2/docs/develop-plugin.md:107]: if it is a local pulsar cluster, should we need to configure the `standalone.conf` file?

**Review by Huanli-Meng — COMMENTED** [review]
Leave some comments. PTAL

File: site2/docs/develop-plugin.md (hunk 9 above)
Comment (Anonymitaet) [reply site2/docs/develop-plugin.md:107]: yes, added

File: site2/docs/develop-plugin.md (hunk 7 above)
Comment (Anonymitaet) [reply site2/docs/develop-plugin.md:50]: A consumer sends a flow permit request to a broker to get messages.  https://pulsar.apache.org/docs/en/develop-binary-protocol/#flow-control

**Comment by Anonymitaet** [comment]
@Huanli-Meng incorporated your comments, PTAL

**Review by Huanli-Meng — APPROVED** [review]
LGTM

**Pull request merged**